src/components/data-grid/DataGrid.tsx

```diff
@@ -128,22 +154,37 @@ export const DataGrid = ({
     gridApi.sizeColumnsToFit();
   };
 
-  const onSelectView = (view: DataGridView | null) => {
-    setSelectedView(view);
-    setViewState(view?.viewState!);
+  const handleActivateView = async (id: string) => {
+    const view = allViews?.find((view) => view.id === id);
+    if (view) {
+      setViewState(view?.viewState!);
+
+      if (view.id && onActivateView) {
+        await onActivateView(view.id);
+      } else if (onDeactivateView) {
+        const activeView = allViews.find((view) => view.active);
+        if (activeView) {
+          await onDeactivateView(activeView.id);
+        }
+
+        view.active = true;
```

Comment:
I would move

```
view.active = true;
setAllViews([...allViews.filter((x) => x.id !== id), view]);
```

 to right after:

```
setViewState(view?.viewState!);
```

to make sure there is no delay between activating a view and seeing the button change

Reply:
Fixed